commons/src/main/java/com/powsybl/commons/config/PlatformConfigNamedProvider.java

```diff
@@ -0,0 +1,197 @@
+            return find(name, moduleName,
+                    Arrays.asList(DEFAULT_SERVICE_IMPL_NAME_PROPERTY,
+                            LEGACY_SERVICE_IMPL_NAME_PROPERTY),
+                    clazz, platformConfig);
+        }
+
+        private static Optional<String> getOptionalFirstProperty(ModuleConfig moduleConfig,
+                List<String> propertyNames) {
+            return propertyNames.stream().map(moduleConfig::getOptionalStringProperty)
+                    .filter(Optional::isPresent).map(Optional::get).findFirst();
+        }
+
+        @SuppressWarnings("unchecked")
+        private static <K, V, T extends V> T alwaysSameComputeIfAbsent(
+                Map<K, V> map, K key,
+                Function<? super K, T> mappingFunction) {
+            // Casting to (T) is safe if we awlays pass the same T argument for a given key
+            return (T) map.computeIfAbsent(key, mappingFunction);
+        }
+
+        private static <T extends PlatformConfigNamedProvider> T find(String name,
+                String moduleName, List<String> propertyNames, Class<T> clazz,
+                PlatformConfig platformConfig) {
+            List<T> providers = alwaysSameComputeIfAbsent(PROVIDERS, clazz,
+                k -> Lists.newArrayList(ServiceLoader.load(clazz)));
+            return find(name, moduleName, propertyNames, providers, platformConfig);
+        }
+
+        // package private for tests
+        static <T extends PlatformConfigNamedProvider> T find(String name,
+                String moduleName, List<String> propertyNames, List<T> providers,
+                PlatformConfig platformConfig) {
+            Objects.requireNonNull(providers);
+            Objects.requireNonNull(platformConfig);
+
+            if (providers.isEmpty()) {
+                throw new PowsyblException("No " + moduleName + " providers found");
+            }
+
+            // if no implementation name is provided through the API we look for information
+            // in platform configuration
+            String finalName = name != null ? name
+                    : platformConfig.getOptionalModuleConfig(moduleName)
+                            .flatMap(mc -> getOptionalFirstProperty(mc, propertyNames))
+                            .orElse(null);
+            T provider;
+            if (providers.size() == 1 && finalName == null) {
+                // no information to select the implementation but only one provider, so we can
+                // use it by default
+                // (that is be the most common use case)
+                provider = providers.get(0);
+            } else {
+                if (providers.size() > 1 && finalName == null) {
+                    // several providers and no information to select which one to choose, we can
+                    // only throw
+                    // an exception
+                    List<String> loadFlowNames = providers.stream()
```

Comment:
Should be renamed?

Reply:
done. I used clazz.getSimpleName() in the message because this is expected to be used on our classes which have very specific names (e.g. LoadFlowProvider)